DeDriftAndResample/MSMRemoveGroupDrift.sh

```diff
@@ -4,488 +4,326 @@
-		log_Msg "p_CommonFolder: ${p_CommonFolder}"
-	fi
+opts_AddMandatory '--group-average-name' 'GroupAverageName' 'name' "output group average name (e.g. S900)"
 
-	if [ -z "${p_GroupAverageName}" ]; then
-		echo "ERROR: group average name (--group-average-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_GroupAverageName: ${p_GroupAverageName}"
-	fi
+opts_AddMandatory '--input-registration-name' 'InRegName' 'string' "Name of the input registration (e.g. MSMAll)"
 
-	if [ -z "${p_InputRegName}" ]; then
-		echo "ERROR: input registration name (--input-registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_InputRegName: ${p_InputRegName}"
-	fi
+opts_AddMandatory '--target-registration-name' 'TargetRegName' 'string' "Name of the target registration (e.g. MSMAll)"
 
-	if [ -z "${p_TargetRegName}" ]; then
-		echo "ERROR: target registration name (--target-registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_TargetRegName: ${p_TargetRegName}"
-	fi
+opts_AddMandatory '--registration-name' 'RegName' 'string' "Name of the registration (e.g. MSMAll)"
 
-	if [ -z "${p_RegName}" ]; then
-		echo "ERROR: registration name (--registration-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_RegName: ${p_RegName}"
-	fi
+opts_AddMandatory '--high-res-mesh' 'HighResMesh' 'numstring' "representing the highres mesh (e.g. 164)"
 
-	if [ -z "${p_HighResMesh}" ]; then
-		echo "ERROR: high resolution mesh (--high-res-mesh=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_HighResMesh: ${p_HighResMesh}"
-	fi
+opts_AddMandatory '--low-res-meshes' 'LowResMesh' 'numstring' "representing the low res mesh (@ delimited list) (e.g. 32)"
 
-	if [ -z "${p_LowResMesh}" ]; then
-		echo "ERROR: low resolution mesh (--low-res-mesh=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_LowResMesh: ${p_LowResMesh}"
-	fi
 
-	if [ ${error_count} -gt 0 ]; then
-		echo "For usage information, use --help"
-		exit 1
-	fi
-}
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
 
-# ------------------------------------------------------------------------------
-#  Show Tool Versions
-# ------------------------------------------------------------------------------
+#display the parsed/default values
+opts_ShowValues
 
-show_tool_versions()
-{
-	# Show HCP pipelines version
-	log_Msg "Showing HCP Pipelines version"
-	"${HCPPIPEDIR}"/show_version --short
+# Verify required environment variables are set and log value
+# log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
 
- 	# Show wb_command version
-	log_Msg "Showing wb_command version"
-	${CARET7DIR}/wb_command -version
-}
 
 # ------------------------------------------------------------------------------
-#  Main processing of script.
+#  Get the command line options for this script.
```

Comment:
This comment is wrong, probably just delete it.

Reply:
Done